source/_drafts/Video-js-5-s-fluid-mode-and-playlist-picker.md

```diff
@@ -0,0 +1,208 @@
+});
+```
+Or you can add one of the fluid classes to the player: `.vjs-fluid`, `.vjs-4-3`, `.vjs-16-9`:
+```html
+<video id="preview-player" class="video-js vjs-fluid" controls data-setup={}>
+```
+`.vjs-4-3` maintains a 4:3 aspect ratio for the video and `.vjs-16-9` maintains a 16:9 one.  `.vjs-fluid` is a bit more special. It waits for the video metadata to load and then uses the video width and video height to calculate the correct aspect ratio to use for the video.
+
+## Playlist picker
+This works great if you only have the player by itself. What if you are trying to a attach a playlist to the video element and keep it at the same height? Like we did on the [advanced example page][advanced-example] on the videojs website. Like so:
+![](videojs-with-playlist.png)
+
+We *could* calculate how much the padding top should be depending on the width of the playlist picker or the container element but then each time a video changes we would need to recalculate the height of the playlist picker. Instead, we can rely on videojs to do all the work.
+
+### Attaching the playlist picker
+For this example, I'm using the [videojs-playlist-ui][playlist-ui] and [videojs-playlist][playlist] plugins for the playlist functionality.
+I then wrap the player in a container and put the playlist-ui element in there as well.
+```html
+<section class="main-preview-player">
+  <video id="preview-player" class="video-js vjs-fluid" controls preload="auto" crossorigin="anonymous">
+    <p class="vjs-no-js">To view this video please enable JavaScript, and consider upgrading to a web browser that <a href="http://videojs.com/html5-video-support/" target="_blank">supports HTML5 video</a></p>
+  </video>
+
+  <ol class="vjs-playlist"></ol>
+</section>
+```
+
+Now we can relatively quickly and make them align together with some CSS:
+```css
+.video-js {
+  width: 70%;
+  float: left;
+}
+.vjs-playlist {
+  width: 30%;
+  float: right;
+}
+```
+Note: this is super simplistic and probably shouldn't be used in real pages
+
+![](videojs-playlist-not-fluid.png)
+
+### How to make it fluid
+As you can see in the preceeding screenshot, it isn't aligned correctly with the player like in the screenshot above.
+Video.js calculates the aspect ratio and then adds a stylesheet to the page:
+```css
+.preview-player-dimensions.vjs-fluid {
+  padding-top: 41.66666666666667%;
+}
+```
+That percentage is because the aspect ratio for our oceans clip is 2.4.
+
+So, to make sure that the playlist picker is the same height, we can just add the player dimensions class to it:
+```html
+<ol class="vjs-playlist preview-player-dimensions vjs-fluid"></ol>
+```
+
+### How to make it line up
+One of the easiest ways of making these two things line up correctly is to use flexbox. It'll let you make the player and playlist picker grow to fill up as much space as needed and shrink down. It'll also let you have the playlist picker collapse underneath the player if the width of the page is too small.
```

Comment:
Hmm, I think if you're using "you" it might flow better to use neither and say "It'll make the player and playlist picker grow to fill up as much space as needed and shrink down."